pytorch_lightning/core/memory.py

```diff
@@ -50,20 +50,31 @@ def get_variable_sizes(self):
         input_ = self.model.example_input_array
 
         if self.model.on_gpu:
-            input_ = input_.cuda(0)
+            device = next(self.model.parameters()).get_device()
```

Comment:
Not very sure about this getting device...
Probably also check if there is a next one?